libs/sdk-core/src/input_parser.rs

```diff
@@ -258,39 +259,51 @@ pub(crate) async fn post_and_log_response(url: &str, body: Option<String>) -> Sd
 /// Makes a GET request to the specified `url` and logs on DEBUG:
 /// - the URL
 /// - the raw response body
-pub(crate) async fn get_and_log_response(url: &str) -> SdkResult<String> {
+/// - the response HTTP status code
+pub(crate) async fn get_and_log_response(url: &str) -> SdkResult<(String, StatusCode)> {
     debug!("Making GET request to: {url}");
 
     let response = get_reqwest_client()?
         .get(url)
         .send()
         .await
         .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?;
-    match response.status().is_success() {
-        true => response
-            .text()
-            .await
-            .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() }),
-        false => {
-            let err = format!(
-                "GET request {} failed with status: {}",
-                url,
-                response.status()
-            );
-            error!("{err}");
-            Err(SdkError::ServiceConnectivity { err })
-        }
-    }
+    let status = response.status();
+    let raw_body = response
+        .text()
+        .await
+        .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?;
+    debug!("Received response, status: {status}, raw response body: {raw_body}");
+
+    Ok((raw_body, status))
 }
 
-/// Wrapper around [get_and_log_response] that, in addition, parses the payload into an expected type
-pub(crate) async fn get_parse_and_log_response<T>(url: &str) -> SdkResult<T>
+/// Wrapper around [get_and_log_response] that, in addition, parses the payload into an expected type.
+///
+/// ### Arguments
+///
+/// - `url`: the URL on which GET will be called
+/// - `enforce_status_check`: if true, the HTTP status code is checked in addition to trying to
+/// parse the payload. In this case, an HTTP error code will automatically cause this function to
+/// return `Err`, regardless of the payload. If false, the result type will be determined only
+/// by the result of parsing the payload into the desired target type.
+pub(crate) async fn get_parse_and_log_response<T>(
+    url: &str,
+    enforce_status_check: bool,
+) -> SdkResult<T>
 where
     for<'a> T: serde::de::Deserialize<'a>,
 {
-    let raw_body = get_and_log_response(url).await?;
+    let (raw_body, status) = get_and_log_response(url).await?;
+    let parse_res = serde_json::from_str::<T>(&raw_body);
```

Comment:
Better to move this line after the enforce_status_check block

Reply:
Done: f9ecc4e8936e3846dcbfba4a3e38672acdc11b36